safe-global/safe-wallet-web
# Fix: don't treat user rejections as errors

## What it solves

Resolves #2701

## How this PR fixes it
When a user rejects a transaction through their connected wallet:
- do not log it as an error to sentry.
- do not send the user an error notification.
- Show `The transaction was rejected by the user.` in the tx execution box instead of a generic error message.

## How to test it
- Create a new transaction.
- When you are prompted to accept the transaction in Metamask, click reject.
- There should not be an error logged to sentry.
- There should be no error notification.
- The Execution box should say `The transaction was rejected by the user.`
- This will effect anywhere the user is prompted to accept or reject a transaction, including:
    - signing a tx
    - executing a tx
    - signing a message
    - batch txs
    - tx with spending limits
    - relayed txs
    - bulk txs
    - account recovery

## Screenshots
Before: 
<img width="1278" alt="Screenshot 2024-01-10 at 09 51 20" src="https://github.com/safe-global/safe-wallet-web/assets/17801424/c0d18f24-8cfb-46b2-9820-1a2ad7c7567a">

After:
<img width="1278" alt="Screenshot 2024-01-10 at 15 36 39" src="https://github.com/safe-global/safe-wallet-web/assets/17801424/e84b530d-9525-45e1-9627-4ed09368ed0a">


## Checklist
